Makefile

```diff
@@ -36,24 +41,41 @@ endif
 
 GO_STATIC_FLAGS=-tags 'netgo,osusergo,static_build'
 
-
 .PHONY: build
 build:
+ifeq ($(LIBGIT2_ONLY),true)
+	docker buildx build \
+		--platform=$(PLATFORMS) \
+		--tag $(IMG):$(TAG) \
+		--file Dockerfile.libgit2-only \
+		$(BUILD_ARGS) .
+else
 	docker buildx build \
 		--platform=$(PLATFORMS) \
 		--tag $(IMG):$(TAG) \
 		--file Dockerfile \
 		$(BUILD_ARGS) .
+endif
 
 .PHONY: test
 test:
+ifeq ($(LIBGIT2_ONLY),true)
+	docker buildx build \
+		--platform=$(PLATFORMS) \
+		--tag $(IMG):$(TAG)-test \
+		--build-arg LIBGIT2_IMG=$(IMG) \
+		--build-arg LIBGIT2_TAG=$(TAG) \
+		--file Dockerfile.test-libgit2-only \
+		$(BUILD_ARGS) .
+else
 	docker buildx build \
 		--platform=$(PLATFORMS) \
 		--tag $(IMG):$(TAG)-test \
 		--build-arg LIBGIT2_IMG=$(IMG) \
 		--build-arg LIBGIT2_TAG=$(TAG) \
 		--file Dockerfile.test \
 		$(BUILD_ARGS) .
+endif
```

Comment:
nit: Both `if`'s here could be replaced by the docker files being a variable. Which could be set in the same block as `IMG` on line 4-6.